### PR TITLE
fix GEARPUMP-2, Define REST API to submit job jar

### PR DESCRIPTION
**Pull request by clockfly**
The Github/Apache Jira integration is not complete yet, here is the jira link:
https://issues.apache.org/jira/browse/GEARPUMP-2

The original discussion can be found at: https://github.com/gearpump/gearpump/issues/1977

Changes:
1. Document how to submit an application jar by REST api.
2. Allow user to compose the config that need to be submitted in javascript. This is done by adding an additional field "configstring".

### CODE REVIEW

File: services/dashboard/services/restapi.js [hunk 1]
```diff
@@ -129,13 +129,16 @@ angular.module('dashboard')
...
+        _submitApp: function(url, files, fileFieldNames, executorNum, args, onComplete) {
           var upload = Upload.upload({
-            url: url + params,
+            url: url,
             method: 'POST',
             file: files,
-            fileFormDataName: formFormNames
+            fileFormDataName: fileFieldNames,
+            fields: {
+              "executornumber": executorNum,
```
Comment (stanleyxu2005) [inline services/dashboard/services/restapi.js]: Maybe the correct name would be `executorCount`. And please update all occurrence, if the rename makes sense to you.

File: services/jvm/src/main/scala/io/gearpump/services/MasterService.scala [hunk 2]
```diff
@@ -188,6 +194,22 @@ class MasterService(val master: ActorRef,
...
     }
   }
+
+  private def mergeConfig(configFile: Option[File], configString: String): Option[File] = {
+    if (configString == null || configString.isEmpty) {
+      configFile
+    } else {
+      configFile match {
+        case Some(file) =>
+          Files.write(file.toPath, ("\n" + configString).getBytes(UTF_8), APPEND)
```
Comment (stanleyxu2005) [inline services/jvm/src/main/scala/io/gearpump/services/MasterService.scala]: config file is HOCON and config string is JSON. Does appending really work?

File: services/dashboard/services/restapi.js (hunk 1 above)
Comment (clockfly) [inline services/dashboard/services/restapi.js]: It make sense, thanks.

File: services/jvm/src/main/scala/io/gearpump/services/MasterService.scala (hunk 2 above)
Comment (clockfly) [inline services/jvm/src/main/scala/io/gearpump/services/MasterService.scala]: config string is also in HOCON.

And it works even if we directly combine HOCON + JSON.

**Comment by stanleyxu2005** [comment]
Lgtm +1

**Comment by clockfly** [comment]
@karol-brejna-i 

The doc style has been fixed.

**Comment by whjiang** [comment]
Are we still using github/gearpump/gearpump repo?

**Comment by clockfly** [comment]
@whjiang 

The code import to Apache git is still in process (blocked at Apache infrastructure), it is tracked by https://issues.apache.org/jira/browse/INFRA-11435

**Comment by clockfly** [comment]
Before the code import is complete, we still use the current repo.

**Pull request merged**